packages/terra-application-layout/src/header/_ApplicationHeader.jsx

```diff
@@ -0,0 +1,262 @@
+    });
+  }
+
+  handleUtilityPopupCloseRequest() {
+    if (this.state.utilityComponent) {
+      this.setState({ utilityComponent: undefined });
+    }
+  }
+
+  handleUtilityOnChange(event, itemData) {
+    const { utilityConfig, app } = this.props;
+
+    utilityConfig.onChange(event, itemData, app && app.disclose);
+  }
+
+  renderToggle() {
+    const { layoutConfig, intl } = this.props;
+
+    if (layoutConfig.toggleMenu) {
+      return (
+        <div className={cx('toolbar-toggle')}>
+          <button
+            className={cx('toggle-button')}
+            aria-label={intl.formatMessage({ id: 'Terra.applicationLayout.applicationHeader.menuToggleLabel' })}
+            onClick={layoutConfig.toggleMenu}
+            data-application-header-toggle
+          >
+            <IconMenu />
+          </button>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderAppName() {
+    const { nameConfig } = this.props;
+
+    if (nameConfig && (nameConfig.accessory || nameConfig.title)) {
+      return (
+        <ApplicationHeaderName accessory={nameConfig.accessory} title={nameConfig.title} />
+      );
+    }
+
+    return null;
+  }
+
+  renderNavigation(isCompact) {
+    const { applicationLinks } = this.props;
+
+    if (!isCompact) {
+      if (applicationLinks.links && applicationLinks.links.length) {
+        return (
+          <ApplicationTabs {...applicationLinks} />
+        );
+      }
+
+      return null;
+    }
+
+    /**
+     * When compact, the navigation region of the header renders the application name component instead. At compact
+     * sizes, the logo region within the ApplicationHeaderLayout is too small to use, so we instead render within
+     * the navigation region which renders with a larger width.
+     */
+    return this.renderAppName();
+  }
+
+  renderExtensions(isCompact) {
+    const { app, layoutConfig, extensions } = this.props;
+
+    if (!isCompact && extensions) {
+      return React.cloneElement(extensions, { app, layoutConfig });
+    }
+
+    return null;
+  }
+
+  renderUtilities(isCompact) {
+    const { utilityConfig } = this.props;
+
+    if (!isCompact && utilityConfig) {
+      return (
+        <ApplicationHeaderUtility
+          onChange={this.handleUtilityOnChange}
+          onDisclose={this.handleUtilityDiscloseRequest}
+          title={utilityConfig.title}
+          accessory={utilityConfig.accessory}
+          menuItems={utilityConfig.menuItems}
+          initialSelectedKey={utilityConfig.initialSelectedKey}
+          data-application-header-utility
+        />
+      );
+    }
+
+    return null;
+  }
+
+  renderUtilitiesPopup() {
+    const { utilityComponent } = this.state;
+
+    if (utilityComponent) {
+      return (
+        <Popup
+          attachmentBehavior="none"
+          contentAttachment="top center"
+          contentHeight="auto"
+          contentWidth="240"
+          isArrowDisplayed
+          isHeaderDisabled
+          isOpen
```

Comment:
bump

Reply:
Children are required, so unless we want to push a blank div into the occupying space we are going to run into warnings at minimum.